src/applications/disability-benefits/526EZ/actions/index.js

```diff
@@ -1,64 +1,138 @@
+export const RESET_PRESTART_STATUS = 'RESET_PRESTART_STATUS';
+export const RESET_PRESTART_DISPLAY = 'RESET_PRESTART_DISPLAY';
+
+export const PRESTART_STATUSES = {
+  notAttempted: 'not-attempted',
   pending: 'pending',
-  incomplete: 'incomplete'
-});
+  none: 'none',
+  expired: 'expired',
+  created: 'created',
+  retrieved: 'retrieved',
+  renewed: 'renewed',
+  notRetrievedSaved: 'not-retrieved-saved',
+  notRetrievedNew: 'not-retrieved-new',
+  notCreated: 'not-created',
+  notRenewed: 'not-renewed',
+};
+
+export const prestartSuccessStatuses = new Set([PRESTART_STATUSES.created, PRESTART_STATUSES.retrieved, PRESTART_STATUSES.renewed]);
+
+export const prestartFailureStatuses = new Set([PRESTART_STATUSES.notCreated, PRESTART_STATUSES.notRenewed, PRESTART_STATUSES.notRetrievedNew, PRESTART_STATUSES.notRetrievedSaved]);
 
+export const prestartPendingStatuses = new Set([PRESTART_STATUSES.none, PRESTART_STATUSES.expired, PRESTART_STATUSES.pending]);
 
-// TODO: replace mock once ITF endpoint setup
-export function submitIntentToFile(formConfig, onChange) {
-  const { intentToFileUrl, formId, migrations, prefillTransformer } = formConfig;
-  let ITFStatus = ITFStatuses.pending;
-  return dispatch => {
+export function setPrestartStatus(status, data) {
+  return {
+    type: SET_PRESTART_STATUS,
+    status,
+    data
+  };
+}
 
-    onChange(ITFStatus);
+export function resetPrestartStatus() {
+  return {
+    type: RESET_PRESTART_STATUS
+  };
+}
 
-    const delay = (ms) => new Promise((resolve, reject) => {
-      ITFStatus = ITFStatuses.active;
-      onChange({ ITFStatus });
+export function resetPrestartDisplay() {
+  return {
+    type: RESET_PRESTART_DISPLAY
+  };
+}
 
-      // TODO: if the backend handles resubmission, this check can be removed
-      if (ITFStatus === ITFStatuses.active) {
-        setTimeout(resolve, ms);
+export function checkITFRequest(dispatch, hasSavedForm) {
+  return apiRequest(
+    '/intent_to_file',
+    null,
+    ({ data }) => {
+      let status;
+      let expirationDate;
+      const itfList = data.attributes.intentToFile;
+      if (!itfList || (Array.isArray(itfList) && itfList.length === 0)) {
+        status = PRESTART_STATUSES.none;
       } else {
-        const errorMessage = 'Network request failed';
-        onChange({
-          ITFStatus,
-          errorMessage
-        });
-        reject(errorMessage);
+        const activeList = itfList.filter(itf => itf.status === 'active');
+        if (activeList.length === 0) {
+          status = PRESTART_STATUSES.expired;
+          expirationDate = itfList.sort((a, b) => a.expirationDate - b.expirationDate)[0].expirationDate;
+        } else {
+          status = PRESTART_STATUSES.retrieved;
+          expirationDate = activeList[0].expirationDate;
+        }
       }
-    });
+      dispatch(setPrestartStatus(status, expirationDate));
+      return status;
+    },
+    ({ errors }) => {
+      const errorMessage = 'Network request failed';
```

Comment:
This is perhaps misleading. It could have failed for any number of reasons, I think. Looks like it's only used in the sentry log. That error message will be as useful without this, I think.

Reply:
This error messaging was modeled on that of `submitToUrl`, is it better to remove this message? Or is there a more descriptive model to follow than `submitToUrl`?

Reply:
`submitToUrl()` creates a new `XMLHttpRequest` and sets event listeners for the different ways it could fail. These event listeners reject the promise with an error message that describes why the request "failed."

In this `apiRequest` failure callback, we don't have any insight as to _why_ the request failed, so it's probably not universally true that the `Network request failed`.

....well......Kind of. I guess it's _technically_ true that it failed for _some_ reason, but it uses the same language as `submitToUrl` without the same level of detail, which can be misleading if we had to debug this later.

That said, since we don't have any other possibilities (that is, `errorMessage` isn't going to be overridden with any other error message), we can remove it entirely and be safe without it. All we know is a certain request failed, and the error we log to sentry should reflect this.